public/app/features/logs/logsModel.ts

```diff
@@ -644,7 +644,19 @@ function defaultExtractLevel(dataFrame: DataFrame): LogLevel {
   try {
     valueField = new FieldCache(dataFrame).getFirstFieldOfType(FieldType.number);
   } catch {}
-  return valueField?.labels ? getLogLevelFromLabels(valueField.labels) : LogLevel.unknown;
+
+  // Attempt to get log level from labels first.
+  if (valueField?.labels) {
+    return getLogLevelFromLabels(valueField.labels);
+  }
+
+  // If the dataframe has a name, attempt to use it as log level.
+  if (dataFrame.name) {
```

Comment:
This does appear to fix the logs volume bug for elastic, at least for me locally, but I'm not sure if making this change in the logsModel is the right place, ideally the dataframe in the elasticsearch datasource would get appended with labels instead.

But, it's not computationally expensive or adding any significant tech debt, so if you add test coverage in the `logsModel.test.ts` there's a good chance this will get approved.

CC @matyax

Reply:
Hey! Yeah, no, this should not be the fix and it introduces this logic for any logging data source with unpredictable consequences. I'll look for the root cause.